File: pdu/python/trajectory_msgs/pdu_cdr_conv_MultiDOFJointTrajectoryPoint.py
from .pdu_pytype_MultiDOFJointTrajectoryPoint import MultiDOFJointTrajectoryPoint
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..builtin_interfaces.pdu_cdr_conv_Duration import *
from ..geometry_msgs.pdu_cdr_conv_Quaternion import *
from ..geometry_msgs.pdu_cdr_conv_Transform import *
from ..geometry_msgs.pdu_cdr_conv_Twist import *
from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_MultiDOFJointTrajectoryPoint(writer: CdrWriter, src: MultiDOFJointTrajectoryPoint):
    writer.write_sequence_length(src.transforms)
    for elem in src.transforms:
        py_to_cdr_body_Transform(writer, elem)
    writer.write_sequence_length(src.velocities)
    for elem in src.velocities:
        py_to_cdr_body_Twist(writer, elem)
    writer.write_sequence_length(src.accelerations)
    for elem in src.accelerations:
        py_to_cdr_body_Twist(writer, elem)
    py_to_cdr_body_Duration(writer, src.time_from_start)


def cdr_body_to_py_MultiDOFJointTrajectoryPoint(reader: CdrReader, dst: MultiDOFJointTrajectoryPoint):
    dst.transforms = []
    for _ in range(reader.read_uint32()):
        elem = Transform()
        cdr_body_to_py_Transform(reader, elem)
        dst.transforms.append(elem)
    dst.velocities = []
    for _ in range(reader.read_uint32()):
        elem = Twist()
        cdr_body_to_py_Twist(reader, elem)
        dst.velocities.append(elem)
    dst.accelerations = []
    for _ in range(reader.read_uint32()):
        elem = Twist()
        cdr_body_to_py_Twist(reader, elem)
        dst.accelerations.append(elem)
    cdr_body_to_py_Duration(reader, dst.time_from_start)
    return dst


def py_to_cdr_MultiDOFJointTrajectoryPoint(src: MultiDOFJointTrajectoryPoint) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_MultiDOFJointTrajectoryPoint(writer, src)
    return writer.bytes()


def cdr_to_py_MultiDOFJointTrajectoryPoint(cdr_payload) -> MultiDOFJointTrajectoryPoint:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = MultiDOFJointTrajectoryPoint()
    return cdr_body_to_py_MultiDOFJointTrajectoryPoint(reader, dst)

File: pdu/javascript/trajectory_msgs/pdu_cdr_conv_MultiDOFJointTrajectoryPoint.js
import { MultiDOFJointTrajectoryPoint } from './pdu_jstype_MultiDOFJointTrajectoryPoint.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Duration } from '../builtin_interfaces/pdu_jstype_Duration.js';
import { PduDurationConverter } from '../builtin_interfaces/pdu_cdr_conv_Duration.js';
import { Quaternion } from '../geometry_msgs/pdu_jstype_Quaternion.js';
import { PduQuaternionConverter } from '../geometry_msgs/pdu_cdr_conv_Quaternion.js';
import { Transform } from '../geometry_msgs/pdu_jstype_Transform.js';
import { PduTransformConverter } from '../geometry_msgs/pdu_cdr_conv_Transform.js';
import { Twist } from '../geometry_msgs/pdu_jstype_Twist.js';
import { PduTwistConverter } from '../geometry_msgs/pdu_cdr_conv_Twist.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduMultiDOFJointTrajectoryPointConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { MultiDOFJointTrajectoryPoint } src
     */
    static to_cdr_body(writer, src) {
        writer.write_sequence_length(src.transforms);
        for (const elem of src.transforms) {
            PduTransformConverter.to_cdr_body(writer, elem);
        }
        writer.write_sequence_length(src.velocities);
        for (const elem of src.velocities) {
            PduTwistConverter.to_cdr_body(writer, elem);
        }
        writer.write_sequence_length(src.accelerations);
        for (const elem of src.accelerations) {
            PduTwistConverter.to_cdr_body(writer, elem);
        }
        PduDurationConverter.to_cdr_body(writer, src.time_from_start);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { MultiDOFJointTrajectoryPoint } dst
     * @returns { MultiDOFJointTrajectoryPoint }
     */
    static cdr_body_to_js(reader, dst) {
        dst.transforms = [];
        for (let i = 0, len = reader.read_uint32(); i < len; i++) {
            dst.transforms.push(PduTransformConverter.cdr_body_to_js(reader, new Transform()));
        }
        dst.velocities = [];
        for (let i = 0, len = reader.read_uint32(); i < len; i++) {
            dst.velocities.push(PduTwistConverter.cdr_body_to_js(reader, new Twist()));
        }
        dst.accelerations = [];
        for (let i = 0, len = reader.read_uint32(); i < len; i++) {
            dst.accelerations.push(PduTwistConverter.cdr_body_to_js(reader, new Twist()));
        }
        PduDurationConverter.cdr_body_to_js(reader, dst.time_from_start);
        return dst;
    }

    /**
     * @param { MultiDOFJointTrajectoryPoint } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { MultiDOFJointTrajectoryPoint }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new MultiDOFJointTrajectoryPoint());
    }
}
